client/cmd/dexc-desktop/app.go

```diff
@@ -0,0 +1,403 @@
+}{
+	windows:  make(map[uint32]*exec.Cmd),
+	zeroLeft: make(chan struct{}, 1),
+}
+
+func closeWindow(windowID uint32) {
+	m := windowManager
+	m.Lock()
+	cmd, found := m.windows[windowID]
+	if !found {
+		m.Unlock()
+		// Probably killed by caller via closeAllWindows.
+		return
+	}
+	delete(m.windows, windowID)
+	remain := len(m.windows)
+	m.Unlock()
+	if remain == 0 {
+		select {
+		case m.zeroLeft <- struct{}{}:
+		default:
+		}
+	}
+	log.Infof("Closing window. %d windows remain open.", remain)
+	cmd.Process.Kill()
+}
+
+func closeAllWindows() {
+	m := windowManager
+	m.Lock()
+	defer m.Unlock()
+	for windowID, cmd := range m.windows {
+		cmd.Process.Kill()
+		delete(m.windows, windowID)
+	}
+}
+
+func runWebview(url string) {
+	w := webview.New(true)
+	defer w.Destroy()
+	w.SetTitle(appTitle)
+	w.SetSize(600, 600, webview.HintMin)
+
+	width, height := defaultWindowWidthAndHeight()
+
+	w.SetSize(width, height, webview.HintNone)
+	w.Navigate(url)
+	w.Run()
+}
+
+func runWebviewSubprocess(ctx context.Context, url string) {
+	cmd := exec.CommandContext(ctx, exePath, "--webview", url)
+	if err := cmd.Start(); err != nil {
+		log.Errorf("webview start error: %v", err)
+		return
+	}
+	m := windowManager
+	m.Lock()
+	m.counter++
+	windowID := windowManager.counter
+	m.windows[windowID] = cmd
+	windowCount := len(m.windows)
+	m.Unlock()
+	defer closeWindow(windowID)
+	log.Infof("Opening new window. %d windows open now", windowCount)
+	if err := cmd.Wait(); err != nil {
+		log.Errorf("webview error: %v", err)
+	}
+}
+
+func systrayOnReady(ctx context.Context, logDirectory string, openC chan<- struct{}, killC chan<- os.Signal) {
+	systray.SetIcon(FavIcon)
+	systray.SetTitle("DEX client")
+	systray.SetTooltip("Self-custodial multi-wallet")
+
+	// TODO: Consider reworking main so we can show the icon earlier?
+	// mStarting := systray.AddMenuItem("Starting...", "Starting up. Please wait...")
+	// var addr string
+	// var ok bool
+	// select {
+	// case addr, ok = <-webserverReady:
+	// 	if !ok { // no webserver started
+	// 		fmt.Fprintln(os.Stderr, "Web server required!")
+	// 		cancel()
+	// 		return
+	// 	}
+	// case <-mainDone:
+	// 	return
+	// }
+
+	// mStarting.Hide()
+
+	mOpen := systray.AddMenuItem("Open", "Open DEX client window")
+	mOpen.SetIcon(SymbolBWIcon)
+	go func() {
+		for range mOpen.ClickedCh {
+			select {
+			case openC <- struct{}{}:
+				log.Debug("Received window reopen request from system tray")
+			default:
+				log.Infof("Ignored a window open request from the system tray")
+			}
+		}
+	}()
+
+	systray.AddSeparator()
+
+	if logDirURL, err := app.FilePathToURL(logDirectory); err != nil {
+		log.Errorf("error constructing log directory URL: %v", err)
+	} else {
+		mLogs := systray.AddMenuItem("Open logs folder", "Open the folder with your DEX logs.")
+		go func() {
+			for range mLogs.ClickedCh {
+				log.Debug("Opening browswer to log directory at", logDirURL)
+				// runWebviewSubprocess(ctx, logDirURL)
```

Comment:
I know this wasn't right, but we can't just comment this out. How about using `browser.OpenFile`?

Reply:
oh, crap. Not intended. Thanks for the review.